lib/dock.rb

```diff
@@ -0,0 +1,54 @@
+class Dock
+  attr_reader     :name,
+                  :max_rental_time
+
+  def initialize(name, max_time)
+    @name            = name
+    @max_rental_time = max_time
+    @active_rentals  = []
+    @active_renters  = []
+    @return_boat     = []
+    @total_charges   = Hash.new(0)
+  end
+
+  def rent(boat, renter)
+    boat.rent_now
+    @active_rentals << boat
+    @active_renters << renter
+  end
+
+  def log_hour
+    @active_rentals.each do |rental|
+      rental.add_hour if rental.rented?
```

Comment:
I don't think this is a big deal, but if a boated is in the active_rentals array, you probably don't need to check if its rented